lib/Doctrine/Common/Annotations/Reader.php

```diff
@@ -86,4 +86,24 @@ function getPropertyAnnotations(\ReflectionProperty $property);
      * @return object|null The Annotation or NULL, if the requested annotation does not exist.
      */
     function getPropertyAnnotation(\ReflectionProperty $property, $annotationName);
+
+    /**
+     * Gets the annotations applied to a constant.
+     *
+     * @param \ReflectionClassConstant $constant The ReflectionClassConstant of the constant
+     *                                           from which the annotations should be read.
+     *
+     * @return array An array of Annotations.
+     */
+    function getConstantAnnotations(\ReflectionClassConstant $constant);
```

Comment:
Adding methods to an existing interface is a BC-break for implementors. Not sure what the right solution is... Maybe create a new interface, and merge it with this one on master

Reply:
Good point. It seems unlikely anybody would implement Reader interface without extending one of the existing implementations, but it is certainly a possibility we need to respect. Therefore, I moved the new methods to a new and deprecated interface `ReaderWithConstantsAnnotations` (which extends `Reader`) and added this interface to all reader implementations. The new interface is already marked as deprecated and removed in the pull request to the 2.0 version. I've intentionally kept redundant `implements Reader` in class declarations to keep the code more readable because the other interface is deprecated and we don't want to confuse anybody.

The v2.0 pull request also contains this interface, but immediately removes it in the next commit, so that it should not cause any trouble if 1.7 branch is ever merged into master.